### DISCUSSION TITLE
upsert generates some SQL with a buggy returning clause

### DISCUSSION THREAD

**Discussion by BrannJoly** [mikro-orm/mikro-orm#4382] [Q&A] [answered]
I have a dbEntity with a ManyToMany relationship defined :

```
@Entity({ tableName: 'projects' })
 {
  @PrimaryKey()
  id: string = uuidv4();

  @Property()
  name: string;

  @ManyToMany(() => UserGroupDbEntity, (group) => group.projects)
  userGroups = new Collection<UserGroupDbEntity>(this);
}
```

And this is UserGroupDbEntity:

@Entity({ tableName: 'user_groups' })
{
export class UserGroupDbEntity extends UserEditableDbEntity<UserGroupId> {
  @ManyToMany(() => UserDbEntity, (user) => user.userGroups, { owner: true })
  users = new Collection<UserDbEntity>(this);

  @ManyToMany(() => ProjectDbEntity, (project) => project.userGroups, { owner: true })
  projects = new Collection<ProjectDbEntity>(this);
}

When trying to insert a new project in the database with upsert, mikroorm adds an erroneous returning clause for userGroups in the sql it executes eg.

```
insert into "public"."projects" 
( "id", "name") 
values ('02fa9e98-9634-4bb7-be16-6f5afe82b58b', 'test ' ) 
on conflict ("id") 
do update set "name" = excluded."name"
returning "userGroups" 
```

Of course this fails with an error because "column "userGroups" does not exist"

I've made sure that my db schema is up to date using mikro-orm migration:create

What could be causing such an issue ?

Thank you
Brann

**Comment by BrannJoly** [comment]
The issue seems to be a regression in the latest version closing [this issue](https://github.com/mikro-orm/mikro-orm/issues/4242)
I think the bug is [here](https://github.com/mikro-orm/mikro-orm/blob/f85f6a305024f1f87cfcad17b141e253e7962556/packages/knex/src/AbstractSqlDriver.ts#LL429C12-L429C21)

**Reply by B4nan** [reply]
Right, I thought the collections are not part of `hydrateProps` but they are... Let me quickly fix that.

**Reply by B4nan** [reply, marked as answer]
https://github.com/mikro-orm/mikro-orm/commit/e342449c1c291b74de8821afc43236f217260165

**Reply by BrannJoly** [reply]
Thank you very much for the quick turnaround !
